isis/src/docsys/documents/DeveloperAddFeature/DeveloperAddFeature.xml

```diff
@@ -0,0 +1,187 @@
+            </p>
+            <p>
+              Depending on how your computer is setup, you may also need to install:
+            </p>
+            <ul>
+              <li> Compiler (Mac OS Xcode, Linux GNU C++)</li>
+              <li> Editor (Vim, Nano, Atom, ...</li>
+            </ul>
+
+            <p>
+              Once you have a complete environment set up for ISIS development, we suggest you 
+              <a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#building-isis3">compile and build</a> 
+              all of ISIS. This step will make sure sure you have everything setup and ready to go.
+            </p>
+
+          <h2><a name="Starting a conversation with other ISIS developers">Starting a conversation with other ISIS developers</a></h2>
+            <p>
+              Before you start writing your new feature, create an <a href="https://github.com/USGS-Astrogeology/ISIS3">issue</a> for 
+              it on the ISIS GitHub site. Doing this will allow other ISIS developers to way in on your preposed feature. By posting
+              a description of what you want to do and asking questions, other developers can give you pointers on how to 
+              implement it.
+            </p>
+
+          <h2><a name="Writing your new application">Writing your new application</a></h2>
+
+            <h3>Creating the files for your new application</h3>
+            <p>
+              The ISIS source code is organized by having a dedicated directory where all of the source code for an application is
+              stored. The files in this directory will be exclusively to your feature. This is where using another application as
+              a template can be very useful.
+            </p>
+            <ol>
+                <li>Decide where your new application belongs</li>
+                <li>Create a directory using the same name as you new application</li>
+                <li>Create a "main.cpp" file</li>
+                <li>Create a "your_new_app.cpp" file</li>
+                <li>Create a "your_new_app.xml" file</li>
+                <li>Create any additional support files (e.g., support_class.cpp, support_class.h</li>
+            </ol>
+            <p>
+              Populate the new files with the code necessary to implement your feature.
+            </p>
+
+            <h3>Compile and link your new application for the first time</h3>
+            <p>
+              Since you are adding a new application to ISIS you will need to re-run the cmake command. This will add 
+              the new application to the list of applications the build system knows about.
+            </p>
+            <ol>
+              <li>Change you current working directory to the ISIS build directory (cd $ISISROOT)</li>
+              <li>Remove the contents of the build directory (rm -rf *)</li>
```

Comment:
We don't need to go this far. You can just remove `CMakeCache.txt` and it should re-do everything but retain the already built files. It'll save a lot of time.

Reply:
Good point